src/autogluon/assistant/ui/tests/test_task_func.py

```diff
@@ -0,0 +1,142 @@
+import subprocess
+from unittest.mock import MagicMock, Mock, patch
+
+import pytest
+from constants import PRESET_MAPPING
+from pages.task import run_autogluon_assistant, setup_local_dataset
+
+
+@pytest.fixture
+def mock_session_state():
+    """Fixture to mock Streamlit session state"""
+    with patch("streamlit.session_state") as mock_state:
+        mock_state.preset = None
+        mock_state.config_overrides = None
+        mock_state.process = None
+        mock_state.pid = None
+        mock_state.output_file = None
+        mock_state.output_filename = None
+        mock_state.task_running = False
+        yield mock_state
+
+
+@pytest.fixture
+def mock_subprocess():
+    """Fixture to mock subprocess.Popen"""
+    with patch("subprocess.Popen") as mock_popen:
+        process_mock = MagicMock()
+        process_mock.pid = 12345
+        process_mock.returncode = 0
+        mock_popen.return_value = process_mock
+        yield mock_popen
+
+
+@pytest.fixture
+def mock_generate_output():
+    """Fixture to mock generate_output_filename"""
+    with patch("pages.task.generate_output_filename") as mock_gen:
```

Comment:
Please use the full path otherwise it won't work once we move into the root tests/unittests/ui/ dir
```suggestion
    with patch("autogluon.assistant.ui.pages.task.generate_output_filename") as mock_gen:
```